### PR TITLE
[storefront] fix customer user country presentation in case of null value

### PR DESCRIPTION
**Pull request by AndrejBlaho**
#### Description, the reason for the PR
there were errors in presenting users billing address if country had null value in https://github.com/shopsys/shopsys/pull/3285 and this PR fixes that

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://ab-nullable-user-data.odin.shopsys.cloud
  - https://cz.ab-nullable-user-data.odin.shopsys.cloud

### CODE REVIEW

File: upgrade-notes/backend_20240806_094817.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#### [framework] address, name and some other user data is now nullable - fix ([#3314](https://github.com/shopsys/shopsys/pull/3314))
```
Comment (vitek-rostislav) [inline upgrade-notes/backend_20240806_094817.md]: I would rather update `upgrade-notes/backend_20240725_061952.md` with something like
```
- see also #project-base-diff for additional fix (https://github.com/shopsys/shopsys/pull/3314)
```

Comment (vitek-rostislav) [reply upgrade-notes/backend_20240806_094817.md]: and fronted upgrade notes should be created here IMHO

Comment (vitek-rostislav) [reply upgrade-notes/backend_20240806_094817.md]: and frontend upgrade notes should be created here IMHO

File: project-base/storefront/utils/user/useCurrentUserContactInformation.ts [hunk 2]
```diff
@@ -102,8 +102,8 @@ const mapCurrentCustomerContactInformationApiData = (
                 : '',
         telephone: apiCurrentCustomerUserData.telephone ?? '',
         country: {
-            value: apiCurrentCustomerUserData.country.code,
-            label: apiCurrentCustomerUserData.country.name,
+            value: apiCurrentCustomerUserData.country ? apiCurrentCustomerUserData.country.code : '',
+            label: apiCurrentCustomerUserData.country ? apiCurrentCustomerUserData.country.name : '',
```
Comment (JanMolcik) [inline project-base/storefront/utils/user/useCurrentUserContactInformation.ts]: Please use optional chaining and nullish coalescing
```js
apiCurrentCustomerUserData.country?.code ?? ''
```

File: project-base/storefront/components/Pages/PersonalData/Detail/PersonalDataDetailContent.tsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ export const PersonalDataDetailContent: FC<PersonalDataDetailContentProps> = ({
                                     </Row>
                                     <Row>
                                         <CellMinor>{t('Country')}</CellMinor>
-                                        <Cell>{userData.country.name}</Cell>
+                                        <Cell>{userData.country ? userData.country.name : ''}</Cell>
```
Comment (JanMolcik) [inline project-base/storefront/components/Pages/PersonalData/Detail/PersonalDataDetailContent.tsx]: Please use optional chaining and nullish coalescing
```js
userData.country?.name ?? ''
```

**Review by JanMolcik — CHANGES_REQUESTED** [review]

**Review by JanMolcik — APPROVED** [review]

**Pull request merged**